cpp/arcticdb/column_store/chunked_buffer.hpp

```diff
@@ -95,8 +96,9 @@ class ChunkedBufferImpl {
     ChunkedBufferImpl() = default;
 
     explicit ChunkedBufferImpl(size_t size) {
-        add_block(size ? size : DefaultBlockSize, 0u);
-        block_offsets_.push_back(0);
+        add_block(size != 0 ? size : DefaultBlockSize, 0u);
+        if(size != 0)
```

Comment:
It didn't break things, it slowed it down, because the chunked buffer does different things depending on the shape of the blocks in the buffer. There is an optimization for only one block, then an integer division and modulus if the blocks are regularly sized (or regularly-sized past the point where the requested offset is), then it does a lower-bound on a vector of offsets.